tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -145,3 +147,245 @@ def test_attack_point_iterator_unrestricted_generator_all_args_len():
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_infinite():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1]
+        }
+    }
+    count = 0
+    for value1, value2 in zip(
+            build_attack_points_iterator(config),
+            itertools.cycle(
+                build_attack_points_iterator(config["configuration"]))):
+        if count > 4:
+            break
+        count += 1
+        assert value1 == value2
+
+
+def test_attack_point_iterator_repeat_infinite_minus_two():
+    config = {
+        "operation": "repeat",
+        "repetitions": -2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2]
+        }
+    }
+    count = 0
+    for value1, value2 in zip(
+            build_attack_points_iterator(config),
+            itertools.cycle(
+                build_attack_points_iterator(config["configuration"]))):
+        if count > 4:
+            break
+        count += 1
+        assert value1 == value2
+
+
+def test_attack_point_iterator_repeat_infinite_no_values():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": []
+        }
+    }
+    output = []
+    for count, value in enumerate(build_attack_points_iterator(config)):
+        if count > 5:
+            break
+        output.append(value)
+
+    assert output == []
+
+
+def test_attack_point_iterator_repeat_infinite_no_values_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": []
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == 0
+
+
+def test_attack_point_iterator_repeat_infinite_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    with pytest.raises(LengthIsInfiniteException):
+        len(build_attack_points_iterator(config))
+
+
+def test_repeat_memory():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1_000_000_000_000,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [[1], [2], [3]]
+        }
+    }
+
+    max_growth_factor = 2
+    python_process = psutil.Process()
+    memory_before_iterator = python_process.memory_info().rss
```

Comment:
nit: consider pre-computing `memory_threshold = memory_before_iterator * max_growth_factor` instead of computing it for each assertion